### PR TITLE
HIP tests for GPU and other build improvements

### PR DESCRIPTION
**Pull request by sonals**
#### Problem solved by the commit

1. Provide a switch for disabling bundling of Alveo drivers with XRT package
2. Fix a build failure on Ubuntu 23.10
3. Add Makefiles for HIP tests to run then on AMD GPU

#### Bug / issue (if any) fixed, which PR introduced the bug, how it was discovered
NA

#### How problem was solved, alternative solutions (if any) and why they were rejected
NA

#### Risks (if any) associated the changes in the commit
NA

#### What has been tested and how, request additional testing if necessary

1. Build tested with default build switches and with -noalveo switch
2. Install tested both debian files.

#### Documentation impact (if any)
NA. This does not change the default behavior of XRT build.

### CODE REVIEW

**Review by maxzhen — APPROVED** [review]
Assuming you have tested the produced package when driver is not bundled and it installs fine. Just wondering if anything needs changing in post install script.

**Comment by keryell** [comment]
*Fix a build failure on Ubuntu 23.10*: I think I fixed this already with https://github.com/Xilinx/XRT/pull/8003

**Comment by sonals** [comment]
> Assuming you have tested the produced package when driver is not bundled and it installs fine. Just wondering if anything needs changing in post install script.

Thanks for reminding, I overlooked the postinst script. Let me validate install/uninstall and update the PR.

**Review by stsoe — CHANGES_REQUESTED** [review]
Can you rebase please?  I think the xrt_graph.cpp change was from #8003 which has been merged.

File: tests/hip/vadd-stream/Makefile.gpu [hunk 1]
```diff
@@ -0,0 +1,52 @@
+# SPDX-License-Identifier: Apache License 2.0 #
+# Copyright (c) 2021-2022 Xilinx, Inc. All rights reserved #
+# Copyright (C) 2022-2023 Advanced Micro Devices, Inc. #
```
Comment (keryell) [inline tests/hip/vadd-stream/Makefile.gpu:3]: Happy new year everywhere! :-)

Comment (sonals) [reply tests/hip/vadd-stream/Makefile.gpu]: Updated

**Comment by sonals** [comment]
> Can you rebase please? I think the xrt_graph.cpp change was from #8003 which has been merged.

Done

**Comment by sonals** [comment]
> > Assuming you have tested the produced package when driver is not bundled and it installs fine. Just wondering if anything needs changing in post install script.
> 
> Thanks for reminding, I overlooked the postinst script. Let me validate install/uninstall and update the PR.

Made updates to postinst script to skip Alveo steps. Install tested both artifacts.

**Comment by maxzhen** [comment]
> > > Assuming you have tested the produced package when driver is not bundled and it installs fine. Just wondering if anything needs changing in post install script.
> > 
> > 
> > Thanks for reminding, I overlooked the postinst script. Let me validate install/uninstall and update the PR.
> 
> Made updates to postinst script to skip Alveo steps. Install tested both artifacts.

Looks good :-)

**Review by chvamshi-xilinx — APPROVED** [review]

**Review by stsoe — APPROVED** [review]

**Pull request merged**